src/app/functions/server/isUserSubscribed.ts

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../api/utils/auth";
+
+type FxaSubscriptionResponse = {
+  subscriptions: Array<{
+    product_id: string;
+    plan_id: string;
+    status: "active";
+  }>;
+};
+
+export const isUserSubscribed = async () => {
+  // Fetch list of subscriptions.
+  let subscriptions: FxaSubscriptionResponse;
+
+  const session = await getServerSession(authOptions);
+  if (!session?.user?.subscriber?.id) {
+    throw new Error("No session");
+  }
+
+  const bearerToken = session?.user.subscriber?.fxa_access_token;
+  if (bearerToken) {
+    const result = await fetch(
+      `${process.env.OAUTH_API_URI}/oauth/mozilla-subscriptions/customer/billing-and-subscriptions`,
```

Comment:
I did talk to subplat folks last week - it turns out there are two slightly different scopes (`profile:subscription` vs. the subscriptions URI that allows calling this API). The former is what should cause the product to show up in the user's profile (JWT).

The only reason we'd need to call these APIs is if we need to be able to differentiate plans (to unlock different features for different plans, which we don't currently have plans for), or if we need more information about subscribers (again no plans for this).

Reply:
> I'm still a bit nervous that we haven't found a good way to do this. Do you know if we can setup SubPlat to pass a query param to Monitor when it redirects the user back to us? That's how we refresh the user's session on Relay, and I think we might possibly be able to solve it by calling [`session.update`](https://next-auth.js.org/getting-started/client#updating-the-session) at that point, combined with a server-side call to update the profile data?

Also yes we get a query param and I will try calling `session.update` and see if that works to refresh the session. Thanks!

Reply:
So I think we should do this separately, because the real underlying problem here is that we have too many test client IDs floating around, and they do not have the proper scopes. This actually works on localhost/dev/stage right now, so I'd like to land as-is and then clean up the client IDs (which might take a few days for SRE), then land a targeted fix that just changes this to use the `subscriptions` claim instead of this API.

Reply:
Oh yeah, I think fixing that is outside of the scope of this PR, but just wanted to see if we could come up with some more plans on how to fix this. Note that I don't expect that `session.update` will do anything unless we also do another [`userinfo` call](https://github.com/mozilla/blurts-server/blob/7107f728e0d2d90af78c45b98e1edf206af86581/src/app/api/utils/auth.ts#L65-L75) and then update the profile.